### ISSUE TITLE
dnn forward Assertion failed

### ISSUE THREAD

**Issue by yuyi1005** [opencv/opencv#10662]
##### System information (version)

- OpenCV => 3.4.0:
- Operating System / Platform =>Windows 64 Bit
- Compiler =>Visual Studio 2015

##### Detailed description

I install opencv with "opencv-3.4.0-vc14_vc15.exe" and try the example in "opencv-3.4.0\sources\samples\dnn\resnet_ssd_face.cpp".

I create a vs2015 project, adding only one cpp file "resnet_ssd_face.cpp". It compiles well, but

The program crashes on Line 110: 
Mat detection = net.forward("detection_out");

Information:
OpenCV Error: Assertion failed (blobs.size() >= 3) in cv::dnn::BatchNormLayerImpl::BatchNormLayerImpl, file C:\build\master_winpack-build-win64-vc14\opencv\modules\dnn\src\layers\batch_norm_layer.cpp, line 29

I also try other models and they don't work either. 

I debug the program, on Line 50:
dnn::Net net = readNetFromCaffe(modelConfiguration, modelBinary);
I find that blobs.size of some layer is correct but some is zero.

Is there anyone know how to solve this problem? Thanks~

**Comment by dkurt** [comment]
@yuyi1005, there is no assertion at line 29 since e3b42bf93b03edc92f4c610c24f0181930e53b18:
https://github.com/opencv/opencv/blob/e3b42bf93b03edc92f4c610c24f0181930e53b18/modules/dnn/src/layers/batch_norm_layer.cpp#L30
Please use http://answers.opencv.org for usage questions and provide ways to reproduce. At least references to model and minimal code sample.

**Comment by dkurt** [comment]
@yuyi1005, can we close it?

**Comment by yuyi1005** [comment]
The example works well. It is because my caffemodel is incomplete. 
But the resnet-101 downloaded from model zoo can not work indeed. Assert failed in full connect layer. 

ResFace101: ResNet-101 for Face Recognition
https://github.com/BVLC/caffe/wiki/Model-Zoo#resface101-resnet-101-for-face-recognition
